cli/execute/execute.go

```diff
@@ -0,0 +1,138 @@
+)
+
+var flags = flag.NewFlagSet("execute", flag.ContinueOnError)
+
+// Bazel-equivalent flags.
+var (
+	target        = flags.String("remote_executor", "grpcs://remote.buildbuddy.io", "Remote execution service target.")
+	instanceName  = flags.String("remote_instance_name", "", "Value to pass as an instance_name in the remote execution API.")
+	timeout       = flags.Duration("remote_timeout", 1*time.Hour, "Timeout used for the action.")
+	remoteHeaders = flag.New(flags, "remote_header", []string{}, "Header to be applied to all outgoing gRPC requests, as a `NAME=VALUE` pair. Can be specified more than once.")
+	actionEnv     = flag.New(flags, "action_env", []string{}, "Action environment variable, as a `NAME=VALUE` pair. Can be specified more than once.")
+)
+
+// Flags specific to `bb execute`.
+var (
+	inputRoot = flags.String("input_root", "", "Input root directory. By default, the action will have no inputs.")
+	// Note: bazel has remote_default_exec_properties but it has somewhat
+	// confusing semantics, so we call this "exec_properties" to avoid
+	// confusion.
+	execProperties = flag.New(flags, "exec_properties", []string{}, "Platform exec property, as a `NAME=VALUE` pair. Can be specified more than once.")
+)
+
+const (
+	usage = `
+usage: bb execute [ options ... ] -- <executable> [ args ... ]
+
+Runs a remote execution request against a remote execution service backend
+using a given command as input.
+
+Args that modify execution can be placed before '--', and the command executable
+and arguments should come afterwards.
+
+Example of running a simple bash command:
+  $ bb execute -- bash -c 'echo "Hello world!"'
+
+Example of running a bash command with runner recycling:
+  $ bb execute --exec_properties=recycle-runner=true -- bash -c 'echo "Runner uptime:" $(uptime)'
+`
+)
+
+func HandleExecute(args []string) (int, error) {
+	cmd, idx := arg.GetCommandAndIndex(args)
+	if cmd != "execute" {
+		return -1, nil
+	}
+	args, cmdArgs := arg.SplitExecutableArgs(args)
+	if err := arg.ParseFlagSet(flags, args[idx+1:]); err != nil {
+		if err == flag.ErrHelp {
+			log.Print(usage)
+			log.Print("\nAll options:")
+			flags.SetOutput(os.Stderr)
+			flags.PrintDefaults()
+			return 1, nil
+		}
+		return -1, err
+	}
+	if len(flag.Args()) > 0 {
+		log.Print("error: command executable and arguments must appear after arg separator '--'")
+		log.Print(usage)
+		return 1, nil
+	}
+	if len(cmdArgs) == 0 {
+		log.Print("error: must provide arg separator '--' followed by command")
+		log.Print(usage)
+		return 1, nil
+	}
+	if err := execute(cmdArgs); err != nil {
+		return -1, err
+	}
+	return 0, nil
+}
+
+func execute(cmdArgs []string) error {
+	ctx := context.Background()
+	md, err := mdutil.Parse(*remoteHeaders...)
+	if err != nil {
+		return err
+	}
+	ctx = metadata.NewOutgoingContext(ctx, md)
+
+	conn, err := grpc_client.DialSimple(*target)
+	if err != nil {
+		return err
+	}
+	env := real_environment.NewRealEnv(nil /*=healthchecker*/)
```

Comment:
we should prolly add a NewBatchEnv() method that returns a stubbed out healthchecker for tools

Reply:
Done - I just set it to `nil` for now since a stub impl is not needed yet.